### PR TITLE
Respect exemption status in summary screens and emails

### PR DESCRIPTION
**Pull request by TThurston**
Exemptions have various statuses (pending, active, revoked, ceased, expired).  Screens which show a list of exemptions, and emails sent to the customer, need to take account of each exemption's status.

### CODE REVIEW

**Comment by TThurston** [comment]
Note: tests for this functionality appear in the Back Office project (alongside comparable tests).

File: app/models/enrollment.rb [hunk 1]
```diff
@@ -142,10 +142,38 @@ def self.not_submitted
...
+  # Returns all exemptions associated with this enrollment, regardless of their status.
   def sorted_enrollment_exemptions
     enrollment_exemptions.preload(:exemption).sort
   end
 
+  # Returns only the exemptions that should be shown on a "review"-type screen, where
+  # the user must confirm that they are registering the correct set of exemptions.
+  def sorted_enrollment_exemptions_for_review_screen
+    enrollment_exemptions.where(status: [
+      WasteExemptionsShared::EnrollmentExemption.statuses[:draft],
```
Comment (lewispb) [inline app/models/enrollment.rb]: Do you need the full module name here, does a simple `::EnrollmentExemption` work?

File: app/views/waste_exemptions_shared/enrollment_mailer/confirmation.html.erb [hunk 2]
```diff
@@ -41,11 +41,11 @@
 </p>
 
 <% @review_data_sections.each_with_index do |review_data, i|  %>
+  <% next if review_data.rows.count.zero? %>
```
Comment (lewispb) [inline app/views/waste_exemptions_shared/enrollment_mailer/confirmation.html.erb]: This type of logic would be better whereever `@review_data_sections` is defined. Use enumerable select.

File: app/views/waste_exemptions_shared/enrollment_mailer/confirmation.html.erb [hunk 3]
```diff
@@ -41,11 +41,11 @@
...
   <table>
     <thead>
     <tr>
-      <th><%= review_data.heading %></th>
-      <th>&nbsp;</th>
+      <th colspan="2"><%= review_data.heading %></th>
```
Comment (lewispb) [inline app/views/waste_exemptions_shared/enrollment_mailer/confirmation.html.erb]: 👍

File: app/models/waste_exemptions_shared/enrollment_exemption.rb [hunk 4]
```diff
@@ -53,6 +53,44 @@ def to_s
...
     end
 
+    # Returns the name/type of the exemption, without any status.
+    def exemption_name_text
+      exemption.to_s
+    end
+
+    # Returns the status of the exemption, without any reference to the
+    # type/name of the exemption.
+    def exemption_status_text
```
Comment (lewispb) [inline app/models/waste_exemptions_shared/enrollment_exemption.rb]: This doesn't seem very DRY to me, and it's a super long method. I wonder if it's possible to break this down. It also could do with a unit test in spec/models/waste_exemptions_shared/enrollment_exemption_spec.rb

File: app/views/waste_exemptions_shared/enrollment_mailer/confirmation.html.erb (hunk 2 above)
Comment (TThurston) [inline app/views/waste_exemptions_shared/enrollment_mailer/confirmation.html.erb]: Agreed - thanks.

File: app/models/enrollment.rb (hunk 1 above)
Comment (TThurston) [inline app/models/enrollment.rb]: Seems like we do; `enrollment` is not in the `WasteExemptionsShared` namespace.  I'd welcome other ways to shorten this?

**Comment by TThurston** [comment]
Useful feedback received with good suggestions.  Will close this PR and submit another taking the proposals into account.

**Pull request closed without merging**